scipy/stats/tests/test_hypotests.py

```diff
@@ -61,13 +65,12 @@ def test_epps_singleton_size(self):
     def test_epps_singleton_nonfinite(self):
         # raise error if there are non-finite values
         x, y = (1, 2, 3, 4, 5, np.inf), np.arange(10)
-        assert_raises(ValueError, epps_singleton_2samp, x, y)
-        x, y = np.arange(10), (1, 2, 3, 4, 5, np.nan)
-        assert_raises(ValueError, epps_singleton_2samp, x, y)
+        with assert_raises(ValueError, match='x must not contain nonfinite'):
+            epps_singleton_2samp(x, y, nan_policy='raise')
 
-    def test_epps_singleton_1d_input(self):
-        x = np.arange(100).reshape(-1, 1)
-        assert_raises(ValueError, epps_singleton_2samp, x, x)
+        x, y = np.arange(10), (1, 2, 3, 4, 5, np.nan)
+        with assert_raises(ValueError, match='The input contains nan values'):
+            epps_singleton_2samp(x, y, nan_policy='raise')
```

Comment:
The diff is confusing here. There are three conceptual changes:
* For the test with an infinite value, make sure the error message matches the expected error message
* For the test with `np.nan`, an error should only be raised when `nan_policy='raise'`
* The `test_epps_singleton_1d_input` is no longer appropriate. It is now valid to pass in an array of size `(n, 1)`. Perhaps instead of removing it entirely, we could check that it does the same thing as passing in an array of size `(n,)`.